File: 0118-pascals-triangle/0118-pascals-triangle.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> pascal={{1}};
        for(int i=1;i<numRows;i++){
            vector<int> row;
            row.push_back(1);
            for(int j=0;j<pascal[i-1].size()-1;j++){
                row.push_back(pascal[i-1][j] + pascal[i-1][j+1]);
            }
            row.push_back(1);
            pascal.push_back(row);
        }
        return pascal;
    }
};

File: 0118-pascals-triangle/0118-pascals-triangle.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> pascal = new ArrayList<>();
        pascal.add(List.of(1));
        for(int i=2;i<=numRows;i++){
            List<Integer> nextRow = new ArrayList<>();
            nextRow.add(1);
            int lastRow = pascal.size()-1;
            for(int j=0;j<pascal.get(lastRow).size()-1;j++){
                nextRow.add(pascal.get(lastRow).get(j) + pascal.get(lastRow).get(j+1));
            }
            nextRow.add(1);
            pascal.add(nextRow);
        }
        return pascal;
    }
}